lib/event_store_http_client/events/write.rb

```diff
@@ -14,14 +14,12 @@ class Write
         dependency :settings
         dependency :client
 
-        def initialize(type, data, stream_name, version)
```

Comment:
Well, I wont' know anything about the event to be written (the type, data, stream_name, OR version) until execution time - this is not available at configuration-time. 

I was a bit troubled by this as well.